test/smoke-asan/hip-dynamic-indirect-lbo/hip-dynamic-indirect-lbo.cpp

```diff
@@ -0,0 +1,51 @@
+#define __HIP_PLATFORM_AMD__ 1
+#include <hip/hip_runtime.h>
+
+#define SHMEM_SIZE 32
+
+void printHipError(hipError_t error) {
+  printf("Hip Error: %s\n", hipGetErrorString(error));
+}
+
+bool hipCallSuccessfull(hipError_t error) {
+  if (error != hipSuccess)
+    printHipError(error);
+  return error == hipSuccess;
+}
+
+__device__ void TestFunction(int *d_ptr){
```

Comment:
Should we add a noinline attribute here?

Reply:
It will remove one CHECK-LINE, but it is fine.
Done.

Reply:
It will remove one CHECK-LINE, but it is fine.
Done.